src/Chainweb/Sync/WebBlockHeaderStore.hs

```diff
@@ -282,15 +282,13 @@ getBlockHeaderInternal headerStore payloadStore candidateHeaderCas candidatePayl
         --
         (maybeOrigin', header) <- casLookup candidateHeaderCas k' >>= \case
             Just !x -> return (maybeOrigin, x)
-            Nothing -> casLookup (_webBlockHeaderStoreCas headerStore) k >>= \case
-                Just (ChainValue _ !x) -> return (Nothing, x)
-                Nothing -> pullOrigin k maybeOrigin >>= \case
-                  Nothing -> do
-                      t <- queryBlockHeaderTask k
-                      pQueueInsert queue t
-                      (ChainValue _ !x) <- awaitTask t
-                      return (Nothing, x)
-                  Just !x -> return (maybeOrigin, x)
+            Nothing -> pullOrigin k maybeOrigin >>= \case
```

Comment:
Will `pullOrigin` check in the local store before a remote query? The local check should be done before any remote query.